### DISCUSSION TITLE
Are textures automatically destroyed?

### DISCUSSION THREAD

**Discussion by Mubelotix** [gfx-rs/wgpu#1580] [Q&A] [answered]
Hey!
Do I have to manually call `destroy` on a `Texture` or will it be dropped automatically when all references to this texture are dropped?
(references are `TextureView`s embeded in `BindGroup`s)

**Comment by cwfitzgerald** [comment, marked as answer]
Correct! Things will be destroyed when all things referencing it die and the gpu finishes with it.

**Reply by Mubelotix** [reply]
Amazing!

**Reply by Yamakaky** [reply]
More generally, does it apply all resource references like BufferSlice, TextureView and BindGroup, not just for textures?

**Reply by cwfitzgerald** [reply]
It does.

**Reply by Yamakaky** [reply]
That's so nice!
